modules/services/secret-service.nix

```diff
@@ -0,0 +1,194 @@
+{ pkgs, lib, config, ... }:
+with lib;
+let cfg = config.services.secret-service;
+in {
+  meta.maintainers = [ maintainers.zebreus ];
+
+  options = {
+    services.secret-service = {
+      enable = mkEnableOption ''
+        Enable managing passwords in the dbus secret service.
+      '';
+
+      secrets = mkOption {
+        default = { };
+        description = ''
+          List of secrets
+        '';
+        type = types.listOf (types.submodule ({ lib, name, ... }:
+          with lib; {
+            options = {
+              label = mkOption {
+                type = types.str;
+                default = name;
+                description = ''
+                  The label of this password. Will be displayed to the user.
+                '';
+              };
+              secretCommand = mkOption {
+                type = types.str;
+                description = ''
+                  The command to run to get the secrect. The command will get run once on every session start and the result will be stored in the secret service.
+                '';
+              };
+              attributes = mkOption {
+                type = types.attrsOf types.str;
+                default = { };
+                description = ''
+                  Attributes for the secret.
+                '';
+              };
+            };
+          }));
+      };
+    };
+
+  };
+
+  config = lib.mkIf cfg.enable {
+    systemd.user = {
+      startServices = mkDefault "sd-switch";
+      services.manage-secret-service-passwords = let
+        storeSecretCommands = lib.concatStringsSep "\n" (builtins.map
+          ({ label, secretCommand, attributes }:
+            let
+              attributesToString = attrs:
+                lib.concatStringsSep " "
+                (lib.mapAttrsToList (name: value: "'${name}' '${value}'")
+                  attrs);
+            in ''
+              # shellcheck disable=all
+              ${secretCommand} | sed -z '$ s/\\n$//' | storeManagedSecret '${label}' ${
+                attributesToString attributes
+              }
+            '') cfg.secrets);
+
+        sharedFunctions = ''
```

Comment:
My preference is to have this in another file, so it's easier to run bash tooling on it (shellcheck, editor syntax) etc. Not mandatory though

Reply:
Sounds good. I moved the helper functions into a `helpers.bash` file (shellcheck prefers the .bash extension over .sh) and created a new directory for it and the nix module.